### PR TITLE
Rename from Markdown React to MDX

### LINKED ISSUE #280: Rename `Markdown React` language to `MDX`
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Problem

Currently the language for MDX files is called `Markdown React` (id: `mdx`). This name, but not id, is consistent the the VSCode builting languages `JavaScript React` (id: `javascriptreact`) and `TypeScript React` (id: `typescriptreact`), which represent `.jsx` and `.tsx` files.

Personally I don’t like that name. It suggests the language is bound to React, but there are more JSX implementations. Also the id isn’t consistent with the name.

### Solution

Rename the language name to `MDX`.

### Alternatives

Keep it as-is.

Not an alternative IMO is renaming the language id, because other extensions already rely on it.

### PR DESCRIPTION
**Pull request by remcohaszing**
This only affects how it is displayed in the UX. It doesn’t change functionality.

This also fixes the link to the tmlanguage JSON schema.

Closes #280

### CODE REVIEW

**Review by ChristianMurphy — APPROVED** [review]
Thanks @remcohaszing !

File: packages/vscode-mdx/package.json [hunk 1]
```diff
@@ -25,8 +25,7 @@
   "main": "out/extension.js",
   "keywords": [
     "mdx",
-    "Markdown",
-    "Markdown React"
+    "markdown"
```
Comment (remcohaszing) [inline packages/vscode-mdx/package.json:28]: These tags show up in the [marketplace](https://marketplace.visualstudio.com/items?itemName=unifiedjs.vscode-mdx). There’s a max of 5 keywords. I don’t mind keeping `Markdown React`, but I dont value that specific keyword either. This field pretty arbritraty, I suggest other keyword suggestions. Should `MDX` be capitalized?

File: packages/vscode-mdx/package.json [hunk 2]
```diff
@@ -70,7 +69,6 @@
       {
         "id": "mdx",
         "aliases": [
-          "Markdown React",
           "MDX"
```
Comment (remcohaszing) [inline packages/vscode-mdx/package.json:72]: It turns out the first alias in this list is used as a display name in VSCode.

File: packages/vscode-mdx/syntaxes/mdx.tmLanguage.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
-  "$schema": "https://raw.githubusercontent.com/Septh/tmlanguage/master/tmLanguage.schema.json",
-  "name": "Markdown React",
+  "$schema": "https://raw.githubusercontent.com/Septh/tmlanguage/master/tmlanguage.json",
+  "name": "MDX",
```
Comment (remcohaszing) [inline packages/vscode-mdx/syntaxes/mdx.tmLanguage.json:3]: I don’t know if / how this name is used, but most tmLanguage definitions seem to define it.

Comment (wooorm) [reply packages/vscode-mdx/syntaxes/mdx.tmLanguage.json:3]: Seems optional. And not explained in the grammar. But good to have it

**Review by wooorm — APPROVED** [review]

**Comment by remcohaszing** [comment]
This has been released in [v1.2.0](https://github.com/mdx-js/vscode-mdx/releases/tag/v1.2.0) :tada:

**Pull request merged**